src/Messenger/IndexationRequestHandler.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace JoliCode\Elastically\Messenger;
+
+use Elastica\Document;
+use JoliCode\Elastically\Client;
+use Psr\Log\NullLogger;
+use Symfony\Component\Messenger\Exception\UnrecoverableMessageHandlingException;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+abstract class IndexationRequestHandler implements MessageHandlerInterface
+{
+    const OP_INDEX = 'index';
+    const OP_DELETE = 'delete';
+    const OP_UPDATE = 'update';
+    const OP_CREATE = 'create';
+
+    const OPS = [
+        self::OP_INDEX,
+        self::OP_DELETE,
+        self::OP_UPDATE,
+        self::OP_CREATE,
+    ];
+
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+
+        // Disable the logs for memory concerns
+        $this->client->setLogger(new NullLogger());
+    }
+
+    public function __invoke(IndexationRequest $message)
+    {
+        $indexer = $this->client->getIndexer();
+        $indexToClass = $this->client->getConfig(Client::CONFIG_INDEX_CLASS_MAPPING);
+        $indexName = array_search($message->getType(), $indexToClass, true);
+
+        if (!$indexName) {
+            throw new UnrecoverableMessageHandlingException(sprintf('The given type (%s) does not exist!', $message->getType()));
+        }
+
+        if (self::OP_DELETE === $message->getOp()) {
+            $indexer->scheduleDelete($indexName, $message->getId());
+            $indexer->flush();
```

Comment:
What if there is alredy something in the buffer? 
I would not flush it here

Reply:
If not here, where should we persist our changes?

Reply:
In the bits of code that call the handler. I apply the same advice with doctrine. I only flush in controller, command, consumer and everything is simpler like that

Reply:
If the handler is called by the workers command (so async), there are no other place to call async.
I'm gonna mark my comment as resolved